### PR TITLE
Add senatorial candidate space and role

### PR DESCRIPTION
**Pull request by kevin-verschaeve**

### CODE REVIEW

File: src/Entity/SenatorialCandidateManagedArea.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    {
+        $this->departmentTags = new ArrayCollection();
+    }
+
+    public function getId(): ?int
+    {
+        return $this->id;
+    }
+
+    public function getDepartmentTags(): iterable
```
Comment (ottaviano) [inline src/Entity/SenatorialCandidateManagedArea.php:50]: what about return the `array` directly here and avoid to call `toArray()` each time?

File: templates/adherent/_list_my_sections.html.twig [hunk 2]
```diff
@@ -76,6 +76,11 @@
         <a href="{{ path('app_municipal_chief_home') }}">Espace Municipales 2020 🇫🇷</a>
     </li>
 {% endif %}
+{% if app.user.isSenatorialCandidate %}
+    <li>
+        <a href="{{ path('app_senatorial_candidate_elected_representatives_list') }}">Espace Senatoriales</a>
```
Comment (ottaviano) [inline templates/adherent/_list_my_sections.html.twig:81]: `Sénatoriale` ?

File: templates/senatorial_candidate/_layout.html.twig [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+{% endblock %}
+
+{% block content %}
+    <main>
+        <section class="manager-space b__nudge--bottom-huge">
+            <header class="manager-header">
+                <div class="l__wrapper">
+                    <div class="first-section">
+                        <div class="el manager-information">
+                            <p>Vous gérez : <span>{{ app.user.senatorialCandidateManagedArea.departmentTags.toArray|join(', ') }}</span></p>
```
Comment (ottaviano) [inline templates/senatorial_candidate/_layout.html.twig:22]: you can avoid `.toArray` call here

File: translations/messages.fr.yml [hunk 4]
```diff
@@ -83,6 +83,7 @@ municipal_manager: Responsable communal
 municipal_manager_supervisor: Responsable attribution
 election_results_reporter: Rapporteur résultats
 print_privilege: Accès LMI
+senatorial_candidate: Candidat Senatoriales 2020
```
Comment (ottaviano) [inline translations/messages.fr.yml:86]: `Sénatoriales` ?

File: translations/messages.fr.yml [hunk 5]
```diff
@@ -764,3 +765,10 @@ delegated_access.form.access.committee: Comités
 delegated_access.created_successfully: L'accès délégué à votre espace a correctement été créé
 delegated_access.updated_successfully: L'accès délégué à votre espace a correctement été mis à jour
 delegated_access.deleted_successfully: L'accès délégué à votre espace a correctement été supprimé
+
+#
+# Senatorial Candidate
+#
+senatorial_candidate.popup.title: Charte de bonne utilisation des outils numériques
+# todo : set correct chart when available
+senatorial_candidate.popup.file_url: https://storage.googleapis.com/en-marche-fr/TECH/charte_numerique_referents.pdf
```
Comment (ottaviano) [inline translations/messages.fr.yml:774]: I will create a copy file with name `charte_numerique_candidat_senatoriales.pdf`, could you use this file please instead? We will need just update the file in the bucket and avoid a new PR + MEP

File: src/Entity/AdherentCharter/SenatorialCandidateCharter.php [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace App\Entity\AdherentCharter;
+
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Entity
```
[the remark this answers is not in the document]
Comment (kevin-verschaeve) [reply src/Entity/AdherentCharter/SenatorialCandidateCharter.php:8]: any of the other charters have this. Should I add it on every charter ?

File: src/Entity/SenatorialCandidateManagedArea.php (hunk 1 above)
Comment (kevin-verschaeve) [reply src/Entity/SenatorialCandidateManagedArea.php:50]: it allows us to benefits from methods like "isEmpty", "filter" etc... Do you want to still return the array from here ?

File: translations/messages.fr.yml (hunk 5 above)
Comment (kevin-verschaeve) [reply translations/messages.fr.yml:774]: yes good point, I update it

File: src/Entity/AdherentCharter/SenatorialCandidateCharter.php (hunk 6 above)
[the remark this answers is not in the document]
Comment (ottaviano) [reply src/Entity/AdherentCharter/SenatorialCandidateCharter.php:8]: yes, please 👍

File: tests/Controller/EnMarche/ElectedRepresentative/ReferentElectedRepresentativeControllerTest.php [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+        $this->assertContains('BOULON Daniel', $crawler->filter('tbody tr.referent__item')->eq(1)->text());
+        $this->assertContains('Conseiller(e) municipal(e) (DIV)Rouen (76000)', $crawler->filter('tbody tr.referent__item')->eq(1)->text());
+        $this->assertContains(' G.s (2018) PS (2014 à 2018)', preg_replace('/\s+/', ' ', $crawler->filter('tbody tr.referent__item')->eq(1)->filter('td')->eq(4)->text()));
+        $this->assertContains('Non', $crawler->filter('tbody tr.referent__item')->eq(1)->text());
+        $this->assertContains('DUFOUR Michelle', $crawler->filter('tbody tr.referent__item')->eq(2)->text());
+        $this->assertContains('Oui', $crawler->filter('tbody tr.referent__item')->eq(2)->text());
+        $this->assertContains('LOBELL', $crawler->filter('tbody tr.referent__item')->eq(3)->text());
+    }
+
+    /** @group debug */
```
Comment (ottaviano) [inline tests/Controller/EnMarche/ElectedRepresentative/ReferentElectedRepresentativeControllerTest.php:41]: could you remove this group?

File: tests/Controller/EnMarche/ElectedRepresentative/SenatorialCandidateElectedRepresentativeControllerTest.php [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+namespace Tests\App\Controller\EnMarche\ElectedRepresentative;
+
+use Liip\FunctionalTestBundle\Test\WebTestCase;
+use Symfony\Component\HttpFoundation\Request;
+use Tests\App\Controller\ControllerTestTrait;
+
+/**
+ * @group functional
+ * @group debug
```
Comment (ottaviano) [inline tests/Controller/EnMarche/ElectedRepresentative/SenatorialCandidateElectedRepresentativeControllerTest.php:11]: same here

**Pull request merged**